addon/mixins/debounced-response.js

```diff
@@ -1,16 +1,19 @@
 import Ember from 'ember';
 
 export default Ember.Mixin.create({
-  debounce: function (handler) {
-    return () => {
-      if (!this.isScheduled) {
-        this.isScheduled = true;
+  debounce(handler) {
+    const self = this;
```

Comment:
I feel kind of silly... Alright I got rid of the `self`s. Also added some tests.